File: src/main/java/Sigmath/computing/Nums.java
package sigmath;



public class Nums
{

    public static int gcd(int a, int b)
    {
        while (b != 0)
        {
            int temp = b;
            b = a % b;
            a = temp;
        }
        return a;
    }

    public static int lcm(int a, int b)
    {
        return Basic.abs(a * b) / gcd(a, b);
    }

    public static int mod(int a, int b)
    {
        int r = a % b;
        return r < 0 ? r + b : r;
    }
}

File: Sigmath.ConvertedToVBNet/computing/nums.vb
Public Class Nums

	Public Shared Function Gcd(a As Integer, b As Integer) As Integer
		While b <> 0
			Dim temp As Integer = b
			b = a Mod b
			a = temp
		End While
		Return a
	End Function

	Public Shared Function Lcm(a As Integer, b As Integer) As Integer
		Return Basic.Abs(a * b) \ Gcd(a, b)
	End Function

	Public Shared Function [Mod](a As Integer, b As Integer) As Integer
		Dim r As Integer = a Mod b
		Return If(r < 0, r + b, r)
	End Function
End Class
